### PR TITLE
add iterator

### PR DESCRIPTION
**Pull request by shuisman**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ const moment = require('moment');
 const crypto = require('crypto');
 const assign = require('lodash.assign');
 const request = require('request');
+const https = require("https");
```
Comment (spirosikmd) [inline src/index.js]: Single quotes :)

File: src/index.js [hunk 2]
```diff
@@ -29,36 +26,56 @@ class Resource {
...
+      host: this.config.host,
+      path: signature.url,
       headers: signature.headers
     };
 
-    return new Promise((resolve, reject) => {
-      request(options, (error, response, body) => {
-        if (!error && response.statusCode == 200) {
-          resolve(JSON.parse(body));
+    https.get(requestOptions,function(res) {
```
Comment (spirosikmd) [inline src/index.js]: Space after comma!

File: src/index.js [hunk 3]
```diff
@@ -29,36 +26,56 @@ class Resource {
...
+    var deferred = Promise.defer();
 
     this.signatureFactory.setUrl(this.url);
     this.signatureFactory.handleQuery(query);
     const signature = this.signatureFactory.sign();
 
-    const options = {
-      url: this.getBaseUrl() + signature.url,
+    var requestOptions = {
+      protocol: 'https:',
```
Comment (spirosikmd) [inline src/index.js]: The config has a protocol defined `protocol: 'https',`, why do you need to define it here as well?

File: src/index.js [hunk 4]
```diff
@@ -29,36 +26,56 @@ class Resource {
...
+          query = assign(query, {params: { since: answer.lastTimestamp }});
+          that.get(query, results).then((results) => {
+            deferred.resolve(results);
+          });
         } else {
-          reject(body);
+          deferred.resolve(results);
         }
       });
     });
```
Comment (spirosikmd) [inline src/index.js]: You could handle the error here as well and reject promise. Take a look [here](https://nodejs.org/api/https.html#https_https_get_options_callback).

```
.on('error', (e) => {
  deferred.reject(e); // Or a string representation of this error
});
```

File: src/index.js [hunk 5]
```diff
@@ -29,36 +26,56 @@ class Resource {
...
    * @param query
    * @returns {Promise}
    */
-  get (query) {
-    if (!query) {
-      query = {};
-    }
+
+  get(query, results) {
+    var query = query || {}
```
Comment (spirosikmd) [inline src/index.js]: Missing `;`

File: src/index.js [hunk 6]
```diff
@@ -29,36 +26,56 @@ class Resource {
    * {
    *   id: '5869485767494'
    *   params: {
-   *     limit: 5
+   *     limit: 5,
+   *     since: 1467964293258
    *   }
    * }
    *
    * @param query
```
Comment (spirosikmd) [inline src/index.js]: Add `@param results` to jsdocs!

File: src/index.js (hunk 3 above)
Comment (shuisman) [inline src/index.js]: This is a legacy setting, and should / is now enforced to https.
Let me remove it from the config in this PR (as it is actually out of scope for this)

File: src/index.js (hunk 4 above)
Comment (shuisman) [inline src/index.js]: interesting

File: src/index.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ const moment = require('moment');
 const crypto = require('crypto');
 const assign = require('lodash.assign');
 const request = require('request');
```
Comment (spirosikmd) [inline src/index.js]: Please remove request as well, also from the package.json! You removed this dependency with using https :)

Comment (shuisman) [inline src/index.js]: done

File: src/index.js (hunk 6 above)
Comment (shuisman) [inline src/index.js]: done

File: src/index.js (hunk 5 above)
Comment (shuisman) [inline src/index.js]: done

File: src/index.js (hunk 2 above)
Comment (shuisman) [inline src/index.js]: done

File: src/index.js (hunk 1 above)
Comment (shuisman) [inline src/index.js]: done

**Comment by spirosikmd** [comment]
Good work @shuisman

**Pull request merged**